env0/resource_template.go

```diff
@@ -22,6 +23,40 @@ func resourceTemplate() *schema.Resource {
 		return nil
 	}
 
+	/*
+	 *	VCS Constraints:
+	 *		GitHub - githubInstallationId
+	 *		GitLab -  gitlabProjectId and tokenId
+	 *		Bitbucket - bitbucketClientKey
+	 *		GH Enterprise - isGitHubEnterprise
+	 *		GL Enterprise - isGitLabEnterprise
+	 *		BB Server - isBitbucketServer
+	 *		Other - tokenId (optional field)
+	 */
+
+	allVCSAttributes := []string{
+		"token_id",
+		"gitlab_project_id",
+		"github_installation_id",
+		"bitbucket_client_key",
+		"is_gitlab_enterprise",
+		"is_bitbucket_server",
+		"is_github_enterprise",
+	}
+
+	allVCSAttributesBut := func(strs ...string) []string {
+		sort.Strings(strs)
+		ret := []string{}
+
+		for _, attr := range allVCSAttributes {
```

Comment:
IMO - with [map](https://stackoverflow.com/questions/38579485/golang-convert-slices-into-map) is more elegant